scipy/ndimage/tests/test_filters.py

```diff
@@ -912,11 +920,99 @@ def test_filter_axes(self, filter_func, extra_args, size0, size, axes, xp):
 
         # result should be equivalent to sigma=0.0/size=1 on unfiltered axes
         axes = xp.asarray(axes)
-        all_sizes = (size if ax in (axes % array.ndim) else size0
-                     for ax in range(array.ndim))
+        all_sizes = tuple(size if ax in (axes % array.ndim) else size0
+                          for ax in range(array.ndim))
         expected = filter_func(array, *extra_args, all_sizes)
         xp_assert_close(output, expected)
 
+    @pytest.mark.parametrize(('filter_func', 'kwargs'),
+                             [(ndimage.laplace, {}),
+                              (ndimage.gaussian_gradient_magnitude,
+                               {"sigma": 1.0}),
+                              (ndimage.gaussian_laplace, {"sigma": 0.5})])
+    def test_derivative_filter_axes(self, xp, filter_func, kwargs):
+        array = xp.arange(6 * 8 * 12, dtype=xp.float64)
+        array = xp.reshape(array, (6, 8, 12))
+
+        # duplicate axes raises an error
+        with pytest.raises(ValueError, match="axes must be unique"):
+            filter_func(array, axes=(1, 1), **kwargs)
+
+        # compare results to manually looping over the non-filtered axes
+        output = filter_func(array, axes=(1, 2), **kwargs)
+        expected = np.empty_like(output)
+        for i in range(array.shape[0]):
+            expected[i, ...] = filter_func(array[i, ...], **kwargs)
+        xp_assert_close(output, expected)
+
+        output = filter_func(array, axes=(0, -1), **kwargs)
+        expected = np.empty_like(output)
+        for i in range(array.shape[1]):
+            expected[:, i, :] = filter_func(array[:, i, :], **kwargs)
+        xp_assert_close(output, expected)
+
+        output = filter_func(array, axes=(1), **kwargs)
+        expected = np.empty_like(output)
+        for i in range(array.shape[0]):
+            for j in range(array.shape[2]):
+                expected[i, :, j] = filter_func(array[i, :, j], **kwargs)
+        xp_assert_close(output, expected)
+
+    @pytest.mark.parametrize(
+        'axes',
+        tuple(itertools.combinations(range(-3, 3), 1))
+        + tuple(itertools.combinations(range(-3, 3), 2))
+        + ((0, 1, 2),))
+    def test_generic_filter_axes(self, xp, axes):
+        array = xp.arange(6 * 8 * 12, dtype=xp.float64)
+        array = xp.reshape(array, (6, 8, 12))
+        axes = xp.asarray(axes)
+        size = 3
+
+        if len(set(axes % array.ndim)) != len(axes):
```

Comment:
it looks like this `set(some_array)` is not portable. We're trying to compare the number of *unique* values in `axes % array.ndim` with the number of values in `axes`, right? Maybe we can use `xp.unique_values`? We may run into https://github.com/data-apis/array-api/issues/834, but that isn't a big deal right now.